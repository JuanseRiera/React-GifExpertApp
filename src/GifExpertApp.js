import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories }) => {
  const [categories, setcategories] = useState(defaultCategories);

  return (
    <>
      <div className="header">
        <h2>GifExpertApp</h2>
        <div>
          <AddCategory setcategories={setcategories} />
        </div>
      </div>

      <hr />

      {categories?.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};

export default GifExpertApp;
