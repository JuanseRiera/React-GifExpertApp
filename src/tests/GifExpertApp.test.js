import React, { Component } from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas de la aplicacion gif expert app", () => {
  test("Se debe mostrar el componente correctamente", () => {
    const categories = ["Sherlock"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
