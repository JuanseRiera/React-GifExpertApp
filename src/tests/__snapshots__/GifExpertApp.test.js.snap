// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de la aplicacion gif expert app Se debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="header"
  >
    <h2>
      GifExpertApp
    </h2>
    <div>
      <AddCategory
        setcategories={[Function]}
      />
    </div>
  </div>
  <hr />
  <GifGrid
    category="Sherlock"
    key="Sherlock"
  />
</Fragment>
`;
